### PR TITLE
Ignoring mocked vkey witnesses during witness serialization

### PR DESCRIPTION
**Pull request by nothingalike**

### CODE REVIEW

**Comment by nothingalike** [comment]
> is this not going to defeat the purpose for calculating tx fees if the mock witnesses aren't serialized into the cbor? maybe a flag or an extension method to strip them out easily?

The mocks are only important to calculate the fee correctly. We dont want them in the final result. That being said, I'm starting to think this will exclude them during the calculate fee process. 

Maybe we have a method to specifically clear out mocked vkey witnesses?

**Comment by nothingalike** [comment]
Ok i must have verified my test incorrectly. i just did another test and my current approach doesnt work. maybe we just make it optional so you can clear them if you want

**Comment by lilpaul** [comment]
maybe it should be a param removeMocks or something like that on the serialization method with a default to false as that would be current default behaviour, but if you know they are in there you can remove them easily during serialization... or just a method to remove them... hmmm... maybe just a method to remove them to start as thats the smallest footprint change and we can always implement the other options later once its been used a bit and we have more experiences to make the judgement from?

File: CardanoSharp.Wallet/TransactionBuilding/TransactionWitnessSetBuilder.cs [hunk 1]
```diff
@@ -69,6 +71,23 @@ public ITransactionWitnessSetBuilder MockVKeyWitness(int count = 1)
             return this;
         }
 
+        public ITransactionWitnessSetBuilder ClearMocks()
```
Comment (lilpaul) [inline CardanoSharp.Wallet/TransactionBuilding/TransactionWitnessSetBuilder.cs:74]: maybe a simple test for this?

File: CardanoSharp.Wallet.Test/TransactionTests.cs [hunk 2]
```diff
@@ -930,11 +932,18 @@ public void MockingWitnessesTest()
...
+            //
+            var fee = transaction.CalculateFee();
+            Assert.Equal(expectedFee, (int)fee);
+            Assert.NotNull(transaction.TransactionWitnessSet);
+            transactionBody.SetFee(fee);
 
-            //assert
-            Assert.Equal("84a50081825820000000000000000000000000000000000000000000000000000000000000000000018182583900477367d9134e384a25edd3e23c72735ee6de6490d39c537a247e1b65d9e5a6498b927f664a2c82343aa6a50cdde47de0a2b8c54ecd9c99c21a000f4240021a000186a0030a0758208dc8a798a1da0e2a6df17e66b10a49b5047133dd4daae2686ef1f73369d3fa16a100818258200000000000000000000000000000000000000000000000000000000000000000584000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000f582a11904d2a1646e616d656e73696d706c65206d65737361676580",
-                serialized.ToStringHex());
+            witnesses.ClearMocks();
```
Comment (nothingalike) [inline CardanoSharp.Wallet.Test/TransactionTests.cs:942]: @lilpaul here is the use of the function in my test. the `Assert.NotNull` and `Assert.Null` are the assertions of this function.

**Review by lilpaul — APPROVED** [review]

**Pull request merged**